connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -3629,6 +3630,64 @@ public void testPollDurationOnSlowConnectorOperations() {
         PowerMock.verifyAll();
     }
 
+    @Test(expected = RejectedExecutionException.class)
+    @SuppressWarnings("unchecked")
+    public void shouldThrowWhenStartAndStopExecutorThrowsRejectedExecutionExceptionAndHerderNotStopping() throws InterruptedException {
+        ExecutorService startAndStopExecutor = EasyMock.mock(ExecutorService.class);
+        herder.startAndStopExecutor = startAndStopExecutor;
+
+        Callable<Void> connectorStartingCallable = () -> null;
+
+        EasyMock.expect(startAndStopExecutor.invokeAll(EasyMock.anyObject(Collection.class))).andThrow(new RejectedExecutionException());
+
+        PowerMock.replayAll(startAndStopExecutor);
+
+        herder.startAndStop(Collections.singletonList(connectorStartingCallable));
+
+    }
+
+    @Test
+    public void shouldHaltCleanlyWhenHerderStartsAndStopsAndConfigTopicReadTimesOut() throws TimeoutException {
+        connectProtocolVersion = CONNECT_PROTOCOL_V1;
+        EasyMock.expect(member.memberId()).andStubReturn("member");
+        EasyMock.expect(member.currentProtocolVersion()).andStubReturn(connectProtocolVersion);
+        final int rebalanceDelayMs = 20000;
+
+        // Assign the connector to this worker, and have it start
+        expectRebalance(Collections.emptyList(), Collections.emptyList(), ConnectProtocol.Assignment.NO_ERROR, 1, singletonList(CONN1), Collections.emptyList(), rebalanceDelayMs);
+
+        member.wakeup();
+        PowerMock.expectLastCall();
+        member.requestRejoin();
+        PowerMock.expectLastCall();
+        member.maybeLeaveGroup(anyString());
+        PowerMock.expectLastCall();
+        worker.stopAndAwaitConnectors();
+        PowerMock.expectLastCall();
+        worker.stopAndAwaitTasks();
+        PowerMock.expectLastCall();
+        member.stop();
+        PowerMock.expectLastCall();
+        configBackingStore.stop();
+        PowerMock.expectLastCall();
+        statusBackingStore.stop();
+        PowerMock.expectLastCall();
+        worker.stop();
+        PowerMock.expectLastCall();
```

Comment:
None of these calls are actually taking place; this becomes apparent when adding `PowerMock::verifyAll` to the end of the test.

```suggestion
```